tests/lottery.spec.ts

```diff
@@ -0,0 +1,249 @@
+    }
+
+    async getCalcPiFromAlpha(provider: ContractProvider, secret: bigint, alpha: Slice): Promise<Slice> {
+        let args = new TupleBuilder();
+        args.writeNumber(secret);
+        args.writeSlice(alpha);
+        const result = await provider.get('ecvrf::rist255::with_secret::prove', args.build());
+        const pi = result.stack.pop();
+        if (pi.type != 'slice') throw new Error('get-method returned invalid value');
+        return pi.cell.beginParse();
+    }
+
+    async sendDeploy(provider: ContractProvider, via: Sender) {
+        await provider.internal(via, {
+            value: toNano('1.0'),
+            body: beginCell().endCell(),
+            bounce: false
+        });
+    }
+
+    async getPublicKey(provider: ContractProvider, secret: bigint): Promise<bigint> {
+        let args = new TupleBuilder();
+        args.writeNumber(secret);
+        return (await provider.get('rist255::get_public_key', args.build())).stack.readBigNumber();
+    }
+
+    async sendWithdraw(provider: ContractProvider, via: Sender) {
+        await provider.internal(via, {
+            value: toNano('0.1'),
+            body: beginCell().storeUint(0xCB03BFAF, 32).endCell(),
+            bounce: true
+        });
+    }
+
+    async getUnfulfilled(provider: ContractProvider): Promise<number> {
+        return (await provider.get('get_unfulfilled', [])).stack.readNumber()
+    }
+}
+
+class LotteryUint implements Contract {
+    constructor(readonly address: Address, readonly init?: { code: Cell; data: Cell }) { }
+
+    static createFromAddress(address: Address, owner1: Slice, owner2: Slice, ecvrf: Slice) {
+        const data = beginCell()
+        .storeUint(0, 64 + 64 + 256 + 256 + 256)
+        .storeBit(1)
+        .storeRef(
+            beginCell()
+            .storeSlice(owner1)
+            .storeSlice(owner2)
+            .storeSlice(ecvrf)
+            .endCell()
+        )
+        .endCell();
+        const init = { code: lotteryCode, data };
+        return new LotteryUint(address, init);
+    }
+
+    async sendDeploy(provider: ContractProvider, via: Sender) {
+        await provider.internal(via, {
+            value: toNano('1.0'),
+            body: beginCell().endCell(),
+            bounce: false
+        });
+    }
+
+    async sendPlayerBet(provider: ContractProvider, via: Sender, value: bigint, player: Slice) {
+        await provider.internal(via, {
+            value: toNano('0.1'),
+            body: beginCell().storeSlice(player).storeCoins(value).endCell(),
+            sendMode: SendMode.PAY_GAS_SEPARATELY
+        });
+    }
+
+    async getLatestRand(provider: ContractProvider): Promise<number> {
+        return (await provider.get('get_random', [])).stack.readNumber()
+    }
+}
+
+describe("lottery test", () => {
+    let blockchain: Blockchain;
+    let ecvrf: SandboxContract<CoordinatorUnit>;
+    let lottery: SandboxContract<LotteryUint>;
+    const secretEcvrf = BigInt(123456);
```

Comment:
You can use `123456n` instead of `BigInt(123456)`. Also applies to all further cases.